src/server/routes/sign.rs

```diff
@@ -66,19 +79,22 @@ impl UnsignedBody {
         Ok(key_manager.verify_bundler_signature(&signature_data, &decoded_signature))
     }
 
-    // FIXME: needs proper error type
     pub async fn sign<KeyManager>(&self, key_manager: &KeyManager) -> Result<String, ()>
     where
         KeyManager: key_manager::KeyManager,
     {
-        let tx_id = self.id.as_bytes().to_vec();
         let bundler_address = key_manager.bundler_address().to_string();
 
         let signature_data = deep_hash(DeepHashChunk::Chunks(vec![
             DeepHashChunk::Chunk(VALIDATOR_AS_BUFFER.into()),
             DeepHashChunk::Chunk(ONE_AS_BUFFER.into()),
-            DeepHashChunk::Chunk(tx_id.into()),
-            DeepHashChunk::Chunk(bundler_address.into()),
+            DeepHashChunk::Chunk(self.id.as_bytes().to_owned().into()),
+            DeepHashChunk::Chunk(self.size.to_string().as_bytes().to_owned().into()),
+            DeepHashChunk::Chunk(self.fee.to_string().as_bytes().to_owned().into()),
+            DeepHashChunk::Chunk(self.currency.as_bytes().to_owned().into()),
+            DeepHashChunk::Chunk(self.block.to_string().as_bytes().to_owned().into()),
+            DeepHashChunk::Chunk(self.validator.as_bytes().to_owned().into()),
+            DeepHashChunk::Chunk(bundler_address.as_bytes().to_owned().into()),
         ]))
```

Comment:
@joshbenaron does this now look correct? Now the validator signature includes also all fields plus adds the `bundler_address`.

Reply:
LGTM